app/views/questionnaire.py

```diff
@@ -125,15 +126,15 @@ def save_questionnaire_store_wrapper(*args, **kwargs):
 @full_routing_path_required
 def get_block(routing_path, eq_id, form_type, collection_id, group_id, group_instance, block_id):  # pylint: disable=unused-argument,too-many-locals
     current_location = Location(group_id, group_instance, block_id)
+    completeness = get_completeness(current_user)
 
-    if not _is_valid_location(routing_path, current_location):
-        return _redirect_to_latest_location(collection_id, eq_id, form_type)
-
-    block = _get_block_json(current_location)
+    route = Router(g.schema, routing_path, completeness, current_location)
```

Comment:
Nitpick - an instance of `Router` is probably still best assigned to variable called `router` rather than `route`. What it returns isn't a route.